allocations/models.py

```diff
@@ -100,3 +104,49 @@ class Charge(models.Model):
 
     def __str__(self):
         return f"{self.allocation.project}: {self.start_time}-{self.end_time}"
+
+
+class ChargeBudget(models.Model):
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="projectbudgets",
+        on_delete=models.CASCADE,
+    )
+    project = models.ForeignKey(
+        Project, on_delete=models.CASCADE
+    )
+    su_budget = models.IntegerField(default=0)
+    enforced_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+
+    class Meta:
+        unique_together = ('user', 'project',)
+
+    def current_usage(self):
+        """
+        Calculate the current charge usage for the user on this project
+        """
+        allocation = su_calculators.get_active_allocation(self.project)
+        if not allocation:
+            return 0
+
+        charges = Charge.objects.filter(allocation=allocation, user=self.user)
+
+        # Avoid doing the calculation if there are no charges
+        if not charges.exists():
+            return 0
+        microseconds_per_hour = 1_000_000 * 3600
+        return charges.annotate(
+            charge_duration=ExpressionWrapper(
```

Comment:
Could this have some comments to explain? It surprises me that this is anything but a sum of all of the charges. I'm especially confused why microseconds per hour comes up.